sensor_msgs/include/sensor_msgs/point_cloud2_iterator.h

```diff
@@ -161,9 +161,23 @@ class PointCloud2Modifier
    * WARNING: THIS FUNCTION DOES ADD ANY NECESSARY PADDING TRANSPARENTLY
    */
   void setPointCloud2FieldsByString(int n_fields, ...);
+
+  class PointFieldInfo{
+    public:
+      PointFieldInfo(std::string name, std::string datatype)
+         : name(name), datatype(datatype){}
+      std::string name;
+      std::string datatype;
+  };
+
+  void addPointCloud2Fields(std::vector<PointFieldInfo> fields);
```

Comment:
Please mirror  the setPointCloud2Fields API here instead of defining a new datatype.